juspay/hyperswitch
# refactor(dynamic_routing): add info logs to log the grpc request and response

## Type of Change



## Description

This PR adds the info logs to log the dynamic routing grpc requests and response objects.

### Additional Changes





## Motivation and Context



## How did you test it?


1. Create merchant_account, api_key and mca (any connector)
2. Toggle dynamic routing

```
curl --location --request POST 'http://localhost:8080/account/merchant_1734609410/business_profile/pro_gPTjn9jLm0CxI8ZD1omL/dynamic_routing/success_based/toggle?enable=metrics' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_K70Wr6zqJLbPECg8eEXrnZ43TW41JNUuXYHzRB50geiG0lViLExjxNUld6sgdNd0' \
--data ''
```

3. Set 100% for dynamic_routing volume split

```
curl --location --request POST 'http://localhost:8080/account/merchant_1734609410/business_profile/pro_gPTjn9jLm0CxI8ZD1omL/dynamic_routing/set_volume_split?split=100' \
--header 'api-key: dev_GFInyC2oIihPleQ4CM8u8WOa02PxN5MHR5W1YAZyXxIYTHt8KDq2VPzRNn26JyJI'
```

4. Create a payment

```
curl --location 'http://localhost:8080/payments' \
--header 'Content-Type: application/json' \
--header 'Accept: application/json' \
--header 'api-key: dev_K70Wr6zqJLbPECg8eEXrnZ43TW41JNUuXYHzRB50geiG0lViLExjxNUld6sgdNd0' \
--data '{
    "amount": 6540,
    "authentication_type": "no_three_ds",
    "confirm": true,
    "currency": "USD",
    "customer_acceptance": {
        "acceptance_type": "online"
    },
    "customer_id": "cus_uYakn3OQTUtAgetLDOE1",
    "payment_method": "card",
    "payment_method_data": {
        "card": {
            "card_cvc": "123",
            "card_exp_month": "10",
            "card_exp_year": "25",
            "card_holder_name": "joseph Doe",
            "card_number": "4242424242424242"
        }
    }
}'
```

Check grafana logs for the payment matching keyword `dynamic_routing_request` and `dynamic_routing_response`

Console request log - 

![image](https://github.com/user-attachments/assets/cadec322-c22d-4288-b87b-c47976a8ac68)

Console response log - 

![image](https://github.com/user-attachments/assets/c0b81c33-70fb-4fae-a7c5-527ff7b1201d)


## Checklist
